src/services/transactions_queued.rs

```diff
@@ -0,0 +1,215 @@
+    context: &Context,
+    safe_address: &String,
+    page_url: &Option<String>,
+    timezone_offset: &Option<String>,
+    trusted: &Option<bool>,
+) -> ApiResult<Page<TransactionListItem>> {
+    let mut info_provider = DefaultInfoProvider::new(context);
+
+    // Parse page meta (offset and limit)
+    let page_meta = PageMetadata::from_url_string(page_url.as_ref().unwrap_or(&"".to_string()));
+    // Adjust the page meta to fetch additional information of adjacent pages
+    let adjusted_page_meta = adjust_page_meta(&page_meta);
+
+    // Allow to also query queued transactions that are not submitted by an owner or delegate
+    let display_trusted_only = trusted.unwrap_or(true);
+
+    // As we require the Safe nonce later we use it here explicitely to query transaction that are in the future
+    let safe_nonce = info_provider.safe_info(safe_address)?.nonce as i64;
+    let url = format!(
+        "{}/v1/safes/{}/multisig-transactions/?{}&nonce__gte={}&ordering=+nonce&trusted={}",
+        base_transaction_service_url(),
+        safe_address,
+        adjusted_page_meta.to_url_string(),
+        safe_nonce,
+        display_trusted_only
+    );
+
+    let body = context
+        .cache()
+        .request_cached(&context.client(), &url, request_cache_duration())?;
+    let mut backend_transactions: Page<MultisigTransaction> = serde_json::from_str(&body)?;
+
+    // If there is a next url we remove the last item for information on the next page
+    let after_tx = if backend_transactions.next.is_some() {
+        backend_transactions.results.pop()
+    } else {
+        None
+    };
+    // Use an iterator to avoid shifting the result vector (would potentially trigger copies)
+    let mut tx_iter = backend_transactions.results.into_iter();
+    // If we are not on the first page then we take the first item to get information on the previous page
+    let before_tx = if page_meta.offset == 0 {
+        None
+    } else {
+        tx_iter.next()
+    };
+    // Nonce of the first item in the next page (-1 if not present)
+    let edge_nonce = after_tx.map_or(-1, |tx| tx.nonce as i64);
+    // Nonce of the last item in the previous page (-1 if not present)
+    let previous_page_nonce = before_tx.map_or(-1, |tx| tx.nonce as i64);
+    let mut last_proccessed_nonce = previous_page_nonce;
+
+    let mut service_transactions: Vec<TransactionListItem> = Vec::new();
+    for (group_nonce, transaction_group) in
+        &tx_iter.group_by(|transaction| transaction.nonce as i64)
+    {
+        // Check if we need to add section headers
+        if last_proccessed_nonce < safe_nonce && group_nonce == safe_nonce {
+            // If the last nonce processed was the initial nonce (-1) and this group nonce is the current Safe nonce then we start the Next section
+            service_transactions.push(TransactionListItem::Label { label: Label::Next })
+        } else if last_proccessed_nonce <= safe_nonce && group_nonce > safe_nonce {
```

Comment:
The first part of this `else if` statement is always `true` , right? What id you check `last_proccessed_nonce == safe_nonce`?

Reply:
no it is not always true ... if you don't have a next transaction on the first page for example